meson.build

```diff
@@ -1,86 +1,177 @@
+
+if host_machine.endian() == 'big'
+  cargs = '-DWORDS_BIGENDIAN=1'
+endif
+
+if (
+  not cc.has_header_symbol('features.h', '_POSIX_C_SOURCE')
+  and cc.get_define('_POSIX_C_SOURCE') == ''
+)
+  cargs += ['-D_POSIX_C_SOURCE=199506L']
+endif
 
 # Check for threadsafe variants of gmtime
-if cc.has_function('gmtime_r', prefix: '#include <time.h>')
+if cc.has_header_symbol(
+  'time.h',
+  'gmtime_r',
+  args: cargs,
+)
   cargs += '-DHAVE_GMTIME_R=1'
-elif cc.has_function('gmtime_s', prefix: '#include <time.h>')
-  cargs += '-DHAVE_GMTIME_S=1'
-endif
+elif cc.has_header_symbol('time.h', 'gmtime_s')
+  if cc.links(
+    '''
+    #include <time.h>
 
-if cc.has_function_attribute('visibility:hidden')
-  cargs +='-DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1'
+    int main() {
+      time_t t;
+      struct tm m;
+      gmtime_s(&m, &t);
+      return 0;
+    }
+    ''',
+    name: 'gmtime_s can be used',
+  )
+    cargs += '-DHAVE_GMTIME_S=1'
+  endif
 endif
 
-if host_machine.system() == 'windows'
-  win = import('windows')
-  lcms2_srcs += win.compile_resources('Projects/VC2019/lcms2.rc')
-  cargs += '-DCMS_DLL_BUILD=1'
+jpeg_dep = dependency(
+  'libjpeg',
+  'jpeg',
+  required: get_option('jpeg'),
+)
+
+if jpeg_dep.found()
+  if (
+    cc.check_header('jconfig.h', dependencies: jpeg_dep)
+    and cc.check_header('jerror.h', dependencies: jpeg_dep)
+    and cc.check_header(
+      'jmorecfg.h',
+      prefix: '#include <jconfig.h>',
+      dependencies: jpeg_dep,
+    )
+    and cc.check_header(
+      'jpeglib.h',
+      prefix: '''
+    #include <stdio.h>
+    #include <stdlib.h>
+    ''',
+      dependencies: jpeg_dep,
+    )
+    and cc.compiles(
+      '''
+    #include <stdio.h>
+    #include <stdlib.h>
+    #include <jpeglib.h>
+    #if JPEG_LIB_VERSION < 62
+    #error IJG JPEG library must be version 6b or newer!
+    #endif
+    ''',
+      dependencies: jpeg_dep,
+      name: 'JPEG library is version 6b or later',
+    )
+  )
+    cargs += '-DHasJPEG=1'
+  else
+    message('JPEG library is not complete -- some components failed test')
+    jpeg_dep = disabler()
+  endif
+else
+  jpeg_dep = disabler()
 endif
 
-liblcms2_lib = library('lcms2', lcms2_srcs,
-  include_directories : inc_dirs,
-  gnu_symbol_visibility: 'hidden',
-  dependencies: deps,
-  c_args: cargs,
-  version: library_version,
-  vs_module_defs: 'src/lcms2.def',
-  install: true,
+tiff_dep = dependency(
+  'libtiff-4',
+  'libtiff',
+  'tiff',
+  required: get_option('tiff'),
 )
 
-liblcms2_dep = declare_dependency(
-  link_with : liblcms2_lib,
-  include_directories : inc_dirs
+if tiff_dep.found()
+  if (
+    cc.check_header('tiff.h', dependencies: tiff_dep)
+    and cc.check_header('tiffio.h', dependencies: tiff_dep)
+    and cc.check_header('tiffconf.h', dependencies: tiff_dep)
+    and cc.has_function('TIFFOpen', dependencies: tiff_dep)
+    and cc.has_function('TIFFClientOpen', dependencies: tiff_dep)
+    and cc.has_function('TIFFIsByteSwapped', dependencies: tiff_dep)
```

Comment:
Do you really need to check that all these functions exist? Are they likely to not exist? How would one get a libtiff without these?

Note that in configure.ac these checks are done only because AC_CHECK_LIB cannot tell if the tiff software is officially installed, and just does the equivalent of Meson's `cc.find_library()` -- so it is important to check that the library in question is the one you meant to find.